packages/data/src/world/mq/gerudo_training_grounds_mq.yml

```diff
@@ -23,40 +19,171 @@
+  locations:
+    "MQ Gerudo Training Grounds Maze Fourth Chest": "true"
+"Gerudo Training Grounds Maze Left Room 6":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Maze Left Room 5": "small_keys_gtg(3)"
+    "Gerudo Training Grounds Maze Center Room": "small_keys_gtg(3)"
+"Gerudo Training Grounds Maze Center Room":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Maze Left Room 6": "small_keys_gtg(3)"
+  events:
+    GTG_ICE_ARROWS_SWITCH: "can_hammer"
+"Gerudo Training Grounds Maze Right Room 1":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Maze Entrance Fork": "true"
+    "Gerudo Training Grounds Maze Right Room 2": "true"
+"Gerudo Training Grounds Maze Right Room 2":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Maze Right Room 1": "true"
+    "Gerudo Training Grounds Lava Room Maze Ledge": "true"
+  locations:
+    "MQ Gerudo Training Grounds Maze Right Side Middle Chest": "true"
+    "MQ Gerudo Training Grounds Maze Right Side Right Chest": "true"
+"Gerudo Training Grounds Right Path First Room":
   dungeon: GTG
   exits:
     "Gerudo Training Grounds": "true"
-    "Gerudo Training Grounds Lava Room": "soul_lizalfos_dinalfos && soul_armos && soul_dodongo && (can_use_sword_master || can_use_sword_goron)"
+    "Gerudo Training Grounds Lava Room Near Beginning": "event(GTG_MQ_RIGHT_FIRST_CLEAR)"
+  events:
+    GTG_MQ_RIGHT_FIRST_CLEAR: "soul_lizalfos_dinalfos && soul_armos && soul_dodongo && (can_use_sword_master || (can_use_sword_goron && has(SWORD_BIGGORON)))"
   locations:
-    "MQ Gerudo Training Grounds Right Side Dinolfos Chest": "soul_lizalfos_dinalfos && soul_armos && soul_dodongo && (can_use_sword_master || can_use_sword_goron)" #This room is too difficult for child to do with other means
+    "MQ Gerudo Training Grounds Right Side Dinolfos Chest": "event(GTG_MQ_RIGHT_FIRST_CLEAR)"
     "MQ Gerudo Training Grounds Wonder Item Dodongo Room Wall": "can_use_bow"
-"Gerudo Training Grounds Lava Room":
+"Gerudo Training Grounds Lava Room Near Beginning":
   dungeon: GTG
   exits:
-    "Gerudo Training Grounds Right Path": "true"
-    "Gerudo Training Grounds Water Room": "silver_rupees_gtg_lava_mq && (can_longshot || can_use_bow || (trick(OOT_MQ_GTG_FLAMES) && (can_hookshot || has_hover_boots)))"
+    "Gerudo Training Grounds Right Path First Room": "true"
+    "Gerudo Training Grounds Lava Room Ledge Before Water Room": "can_longshot || event(GTG_MQ_LAVA_TORCH_ENTRANCE_SIDE) || (trick(OOT_MQ_GTG_FLAMES) && (can_hookshot || has_hover_boots)) || hookshot_anywhere || climb_anywhere"
+    "Gerudo Training Grounds Lava Room Maze Ledge": "(event(GTG_MQ_LAVA_HAMMER_SWITCH) && can_longshot) || (event(GTG_MQ_LAVA_TORCH_ENTRANCE_SIDE) && event(GTG_MQ_LAVA_HAMMER_SWITCH) && can_hookshot) || hookshot_anywhere || climb_anywhere"
+    "Gerudo Training Grounds Lava Room Deep Side Ledge": "hookshot_anywhere || (can_longshot && has_hover_boots && climb_anywhere && event(GTG_LAVA_HOOK_TARGETS))"
```

Comment:
Realized this one will also need the flame circle trick with hookshot anywhere

Reply:
Added.